int8 signal1
int8 signal2

